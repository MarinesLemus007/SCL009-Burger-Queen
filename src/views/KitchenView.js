import React from 'react';
import Navbar from '../components/Navbar';
import OrderTemplate from '../components/OrderTemplate';
import {db} from '../data/firebase';


class KitchenView extends React.Component{
    constructor(props) {
        super(props);
        this.state = {selectedNavbar:'kitchen'}
       this.changeReadyStatus= this.changeReadyStatus.bind(this);
      }


    componentDidMount() {
        db.collection("ordenes").where("not_ready", "==", true).orderBy("time","asc").limit(6).onSnapshot((querySnapshot)=>{
           this.setState({
                data: querySnapshot.docs.map(doc =>{
                   
                    return {data: doc.data()}
                })
            })
               
        });
          
    }

    changeReadyStatus(id) {
        this.state.data.forEach(el=> {
            if (el.data.id === id){
                db.collection("ordenes").doc(id).update({
                    not_ready: false,
                    time_ready:Date.now(),
                    not_delivered: true,
                })
            }
        })
    }

    render(){
        const {data} = this.state;
     
        return (
            <>
            <Navbar state={this.state.selectedNavbar}/>
            <div>
                <p>Pedidos para cocinar</p>
                <OrderTemplate changeReadyStatus={this.changeReadyStatus} data={data}/>
            </div>
            </>
        )
    }
}

export default KitchenView;