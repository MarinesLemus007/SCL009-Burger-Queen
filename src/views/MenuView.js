import React from 'react';
import {db} from '../data/firebase';
import '../index.css';
import Navbar from '../components/Navbar';
import Btn from '../components/Btn'
import Menu from '../data/menu.json'
import Order from '../components/Order';
import OrderName from '../components/OrderName';
import CategoryBtn from '../components/CategoryBtn'
import LunchBtn from '../components/LunchBtn'


class MenuView extends React.Component{
  
  constructor(props){
    super(props)
   
    this.state = {list: [], client:"",selectedNavbar:'menu',errorMsg:''};
    this.add = this.add.bind(this);
    this.delete = this.delete.bind(this);
    this.view = this.view.bind(this);
    this.changeClient = this.changeClient.bind(this)
    this.clearOrder = this.clearOrder.bind(this);
    this.saveOrder = this.saveOrder.bind(this);
    this.addCheeseorEgg = this.addCheeseorEgg.bind(this);
    this.index = 0; // id de cada elemento de orden creado
   
  }
  //aÃ±ade un producto({name,value,id} a la orden(list -> array de objetos [el index comienza en 0 y se utiliza como id]
  add(valueToAdd,nameToAdd){
    let newList = [...this.state.list] 
          for( let i = 0; i < newList.length; i++){ 
            if ( newList[i].name === nameToAdd) {
                newList[i].count++;
                newList[i].value = newList[i].value + valueToAdd;
              return this.setState({list:newList})
            }
          }
    this.setState({list: this.state.list.concat([{name : nameToAdd, unit_value:valueToAdd, value:valueToAdd, id:this.index, count:1 }])});
    this.index ++;
  }

  addCheeseorEgg(eggOrCheese){
    let newList = [...this.state.list] 
    let lastItem = newList[newList.length-1];
    if(lastItem ===undefined){
      return
    }
    if ( eggOrCheese === '+ Queso') {
    
              lastItem.name = lastItem.name+" + Queso"
              lastItem.value = lastItem.value + 500;
              return this.setState({list:newList})
    }
              lastItem.name = lastItem.name+" + Huevo"
              lastItem.value = lastItem.value + 500;
             this.setState({list:newList})
  }
  

  //
  delete(id){
    let newList = [...this.state.list] //crea copia de list para no cambiar directamente el estado de un componente
    for( var i = 0; i < newList.length; i++){ //recorre la newList y consigue la id de cada elemento,si coincide
      if ( newList[i].id === id) { 
        if( newList[i].count !== 1){
          newList[i].value = newList[i].value - newList[i].unit_value ;
          newList[i].count--;
          return this.setState({list:newList})
        }
        newList.splice(i, 1); //con la misma id que se le pasa borra este elemento (desde posicion i, elimina 1 elemento)
        break;
      }
    }
   this.setState({list:newList}) //remplaza la list con la newList modificada
  }


  view(category){ // dependiendo de la categoria cambia el estado, este metodo se pasa como propiedad al boton
    category = category.toLowerCase()
      
    if(category ==="desayunos"){
      this.setState({
        desayunos:true,
        almuerzos:false
        })
    }
    if(category ==="almuerzos"){
      this.setState({
        desayunos:false,
        almuerzos:true
        })
      }
    }

  changeClient(el){
    this.setState({
      client: el.target.value
    })
  }

  clearOrder(){
    this.setState({
      list: [],
      client:"",
      errorMsg:""
    })
  }

  saveOrder(){
    if(this.state.client === ""){
      this.setState({
        errorMsg:'Debe ingresar un nombre.'}) 
      return 
    }
    let idClient =this.state.client + Date.now();

    let data=
    {
    id: idClient,
    client: this.state.client,
    list: this.state.list,
    not_ready: true,
    time: Date.now() 
    }

    db.collection("ordenes").doc(idClient).set(data)
    .then(() => {
        this.clearOrder();
    })

  }



  render(){
    return (
      <>
        <Navbar state={this.state.selectedNavbar} />
          <div className="content-row">
            <section className="button-content-col">
              <div className="category-btn-row">
                {Object.keys(Menu).map(btn=> <CategoryBtn name={btn.toUpperCase()} state={btn === 'Desayunos' ? this.state.desayunos : this.state.almuerzos} view={this.view} key={btn}/>)}
              </div>
                {this.state.desayunos &&
                <div className="item-btn-row"> {Menu.Desayunos.map(btn=><Btn name={btn.name} value={btn.value} add={this.add} key={btn.name}/>)}
                </div> }
                {this.state.almuerzos && <LunchBtn add={this.add} addCheeseorEgg={this.addCheeseorEgg}/>}  
            </section>  
            <aside className="side-content-col">
            <div className="aside-content">
              <div className="line-order">
                <p>PEDIDO</p>
                <p className='error-msg'>{this.state.errorMsg}</p>
              </div>
              
              <OrderName changeClient={this.changeClient} client={this.state.client}/>
              <div className="order-content">
                <Order list = {this.state.list} delete={this.delete}/>
              </div>
              <footer className="footer-side">
                <div className="btn-clear-col">
                  <button className="btn-aside-clear" onClick={this.clearOrder}>LIMPIAR</button>
                </div>
                <div className="btn-send-col">
                  <button className="btn-aside" onClick={this.saveOrder}>ENVIAR </button>
                </div>
              </footer>
              </div>
            </aside>
          </div>
      </>
    )
  }
}

export default MenuView;

