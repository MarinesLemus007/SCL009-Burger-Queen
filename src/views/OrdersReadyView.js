import React from 'react';
import Navbar from '../components/Navbar';
import OrderDeliveryTemplate from '../components/OrderDeliveryTemplate';
import {db} from '../data/firebase';

class OrdersReadyView extends React.Component{
    constructor(props) {
        super(props);
        this.state = {selectedNavbar:'ordersReady'}
        this.changeReadyStatus= this.changeReadyStatus.bind(this);
      }


    componentDidMount() {
        db.collection("ordenes").where("not_delivered", "==", true).onSnapshot((querySnapshot)=>{
            this.setState({
                data: querySnapshot.docs.map(doc =>{       
                    return {data: doc.data()}
                })
            })
        })
    }

    changeReadyStatus(id) {
        this.state.data.forEach(el=> {
            if (el.data.id === id){
                db.collection("ordenes").doc(id).update({
                    not_delivered: false,
                    record: true,
                    time_delivered:Date.now(),
                })
            }
        })
    }

    render(){
        const {data} = this.state;
     
        return (
            <>
            <Navbar state={this.state.selectedNavbar}/>
       
        <div>
        
        <p>Pedidos listos para entregar</p>
              <OrderDeliveryTemplate changeReadyStatus={this.changeReadyStatus} data={data}/>
        
        
        
        </div>
    </>)
    }
}

export default OrdersReadyView;