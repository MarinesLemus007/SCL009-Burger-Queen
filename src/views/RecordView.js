import React from 'react';
import Navbar from '../components/Navbar';
import OrderRecordTemplate from '../components/OrderRecordTemplate';
import {db} from '../data/firebase';

class RecordView extends React.Component{
    constructor(props) {
        super(props);
        this.state = {selectedNavbar:'record'}
       
      }


    componentDidMount() {
        db.collection("ordenes").where("record", "==", true).onSnapshot((querySnapshot)=>{
            this.setState({
                data: querySnapshot.docs.map(doc =>{       
                    return {data: doc.data()}
                })
            })
        })
    }



    render(){
        const {data} = this.state;
     
        return (
            <>
            <Navbar state={this.state.selectedNavbar}/>
       
        <div>
        
        <p>Historial</p>
              <OrderRecordTemplate data={data}/>
        
        
        
        </div>
    </>)
    }
}

export default RecordView;