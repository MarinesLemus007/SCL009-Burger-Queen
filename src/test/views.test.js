import MenuView from '../views/MenuView';
import KitchenView from '../views/KitchenView'
import RecordView from '../views/RecordView'
import OrdersReadyView from '../views/OrdersReadyView'
import React from 'react';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('Componentes existentes en MenuView,KitchenView,RecordView y OrdersReadyView', () => {

it('Deberia encontrar Navbar en MenuView', () => {
    const nameComponent = shallow(<MenuView/>);
    expect(nameComponent.find('Navbar').length).toBe(1);
  });

  it('Deberia encontrar CategoryBtn en MenuView', () => {
    const nameComponent = shallow(<MenuView/>);
    expect(nameComponent.find('CategoryBtn').length).toBe(2);
  });

  it('Deberia encontrar OrderName en MenuView', () => {
    const nameComponent = shallow(<MenuView/>);
    expect(nameComponent.find('OrderName').length).toBe(1);
  });

  it('Deberia encontrar Order en MenuView', () => {
    const nameComponent = shallow(<MenuView/>);
    expect(nameComponent.find('Order').length).toBe(1);
  });

  it('Deberia encontrar Navbar en KitchenView', () => {
    const nameComponent = shallow(<KitchenView/>);
    expect(nameComponent.find('Navbar').length).toBe(1);
  });

  it('Deberia encontrar OrderTemplate en KitchenView', () => {
    const nameComponent = shallow(<KitchenView/>);
    expect(nameComponent.find('OrderTemplate').length).toBe(1);
  });
  
  it('se renderea correctamente el componente RecordView', () => {  
    const component = shallow(<RecordView  />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('se renderea correctamente el componente OrdersReadyView', () => {  
    const component = shallow(<OrdersReadyView  />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('se renderea correctamente el componente MenuView', () => {  
    const component = shallow(<MenuView />);
    expect(toJson(component)).toMatchSnapshot();
  });
});