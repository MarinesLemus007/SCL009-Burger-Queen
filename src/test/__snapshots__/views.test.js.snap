// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componentes existentes en MenuView,KitchenView,RecordView y OrdersReadyView se renderea correctamente el componente MenuView 1`] = `
<Fragment>
  <Navbar
    state="menu"
  />
  <div
    className="content-row"
  >
    <section
      className="button-content-col"
    >
      <div
        className="category-btn-row"
      >
        <CategoryBtn
          className="btnCategory"
          key="Desayunos"
          name="DESAYUNOS"
          view={[Function]}
        />
        <CategoryBtn
          className="btnCategory"
          key="Almuerzos"
          name="ALMUERZOS"
          view={[Function]}
        />
      </div>
    </section>
    <aside
      className="side-content-col"
    >
      <div
        className="aside-content"
      >
        <div
          className="line-order"
        >
          <p>
            PEDIDO
          </p>
          <p
            className="error-msg"
          />
        </div>
        <OrderName
          changeClient={[Function]}
          client=""
        />
        <div
          className="order-content"
        >
          <Order
            delete={[Function]}
            list={Array []}
          />
        </div>
        <footer
          className="footer-side"
        >
          <div
            className="btn-clear-col"
          >
            <button
              className="btn-aside-clear"
              onClick={[Function]}
            >
              LIMPIAR
            </button>
          </div>
          <div
            className="btn-send-col"
          >
            <button
              className="btn-aside"
              onClick={[Function]}
            >
              ENVIAR 
            </button>
          </div>
           
        </footer>
      </div>
    </aside>
  </div>
</Fragment>
`;

exports[`Componentes existentes en MenuView,KitchenView,RecordView y OrdersReadyView se renderea correctamente el componente OrdersReadyView 1`] = `
<Fragment>
  <Navbar
    state="ordersReady"
  />
  <div
    className="order-ready-container"
  >
    <h4>
      PEDIDOS LISTOS PARA ENTREGAR
    </h4>
    <OrderDeliveryTemplate
      changeReadyStatus={[Function]}
    />
  </div>
</Fragment>
`;

exports[`Componentes existentes en MenuView,KitchenView,RecordView y OrdersReadyView se renderea correctamente el componente RecordView 1`] = `
<Fragment>
  <Navbar
    state="record"
  />
  <div
    className="order-record-container"
  >
    <h4>
      HISTORIAL
    </h4>
    <OrderRecordTemplate />
  </div>
</Fragment>
`;
